presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewCandidateExtractor.java

```diff
@@ -33,38 +37,37 @@
     private final Metadata metadata;
     private final Session session;
 
+    Set<QualifiedObjectName> withTables = new HashSet<>();
+
     public MaterializedViewCandidateExtractor(Session session, Metadata metadata)
     {
         this.session = requireNonNull(session, "session is null");
         this.metadata = requireNonNull(metadata, "metadata is null");
     }
 
+    @Override
+    protected Void visitWithQuery(WithQuery node, Void context)
+    {
+        withTables.add(createQualifiedObjectName(session, node, QualifiedName.of(node.getName().toString())));
```

Comment:
Feels like there might be an existing util to create a `QualifiedObjectName` from a `QualifiedName`?